packages/esm-outpatient-app/src/visit-header/work-spaces.ts

```diff
@@ -0,0 +1,192 @@
+/** Adapted from the esm-patient-chart/esm-common-lib/workspaces */
```

Comment:
@donaldkibet in this case I'm fine to write it twice in order to ignore circular dependency problems. Core does not require patient-management to run, nor should it. The most appropriate place for all of the patient management features remains in patient-management. This only duplicates the "start visit" functionality, but I don't expect this logic to change frequently and if it is not working in one space or the other we will quickly notice. Somewhat inelegant but a good solution IMHO

Reply:
You are right @donaldkibet I used the same function from the `common-lib` because I think it is easier for now until we decide to move it to a module that is accessible by the `patient-management`

Reply:
I do think we need to consider moving the "workspace" concept into esm-core, albeit for reasons that aren't specific to this PR. This does add a bit of a weird dependency on @openmrs/patient-chart-app to the outpatient-app, so we haven't entirely avoided the dependency issues (specifically the patient-chart-app needs to be loaded for the i18n to work here).

That said, being pragmatic about this, I think there are a number of things to work through in making the workspace concept more general and this probably isn't the PR to do it.